chains/evm/calls/contract/contract.go

```diff
@@ -0,0 +1,114 @@
+package contract
+
+import (
+	"context"
+	"fmt"
+	"github.com/ChainSafe/chainbridge-core/chains/evm/calls/client"
+	"github.com/ChainSafe/chainbridge-core/chains/evm/calls/consts"
+	"github.com/ChainSafe/chainbridge-core/chains/evm/calls/transactor"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/rs/zerolog/log"
+)
+
+type Contract struct {
+	contractAddress common.Address
+	ABI             abi.ABI
+	bytecode        []byte
+	client          client.ContractCallerDispatcherClient
+	transactor.Transactor
+}
+
+func NewContract(
+	contractAddress common.Address,
+	abi abi.ABI,
+	bytecode []byte,
+	client client.ContractCallerDispatcherClient,
+	transactor transactor.Transactor,
+) Contract {
+	return Contract{
+		contractAddress: contractAddress,
+		ABI:             abi,
+		bytecode:        bytecode,
+		client:          client,
+		Transactor:      transactor,
+	}
+}
+
+func (c *Contract) ContractAddress() *common.Address {
+	return &c.contractAddress
+}
+
+func (c *Contract) PackMethod(method string, args ...interface{}) ([]byte, error) {
+	input, err := c.ABI.Pack(method, args...)
+	if err != nil {
+		return []byte{}, err
+	}
+	return input, nil
+}
+
+func (c *Contract) UnpackResult(method string, output []byte) ([]interface{}, error) {
+	res, err := c.ABI.Unpack(method, output)
+	if err != nil {
+		log.Error().Err(fmt.Errorf("unpack output error: %v", err))
+		return nil, err
+	}
+	return res, err
+}
+
+func (c *Contract) ExecuteTransaction(method string, opts transactor.TransactOptions, args ...interface{}) (*common.Hash, error) {
+	input, err := c.PackMethod(method, args...)
+	if err != nil {
+		log.Error().Err(err)
+		return nil, err
+	}
+	h, err := c.Transact(&c.contractAddress, input, opts)
+	if err != nil {
+		log.Error().Err(err).Msg(method)
+		return nil, err
+	}
+	log.Debug().Str("hash", h.String()).Msgf("%s sent", method)
```

Comment:
Do we want the transactor to log stuff or ExecuteTransaction.
Doesn't really make sense to log hash twice.